### PR TITLE
router: fix router discovery not working after preforming ifup

### PR DESCRIPTION
**Pull request by meurisa**
Tracks interface device index, forces close and reinitialization if the if index has changed
also tracks multicast membership only run add/remove membership once per interface enable/disable

### CODE REVIEW

**Comment by meurisa** [comment]
This commit fixes : https://github.com/openwrt/odhcpd/issues/135

File: src/router.c [hunk 1]
```diff
@@ -81,12 +82,20 @@ int router_setup_interface(struct interface *iface, bool enable)
 		goto out;
 	}
 
+	if (iface->ifindex != iface->current_device_index &&  iface->router_event.uloop.fd >=0 ) {
```
Comment (dedeckeh) [inline src/router.c:85]: This should not be handled in router_setup_interface as there's a specific event NETEV_IFINDEX_CHANGE which can be used to trigger actions if the ifindex changes

**Comment by dedeckeh** [comment]
Besides the hacky way the patch is implemented; it has formal issues like not signed-off and not respecting the existing coding style

**Comment by dedeckeh** [comment]
Issue is fixed in commit https://git.openwrt.org/?p=project/odhcpd.git;a=commit;h=000182fe4f94a5a6ec139456a2b74f0cdea13b9c

**Pull request closed without merging**